File: Projekt1_cpp/Projekt1_cpp/Mlecz.cpp
#include "Mlecz.h"

Mlecz::Mlecz(pair<int, int>pole, Swiat* swiat)
	:Roslina(0, 0, pole, swiat) {}

Organizm* Mlecz::potomek(pair<int, int>pole) {
	return new Mlecz(pole, swiat);
}

void Mlecz::akcja() {
	rozprzestrzenianie();
	rozprzestrzenianie();
	rozprzestrzenianie();
}

void Mlecz::rysowanie() {
	cout << 'm';
}

string Mlecz::getNazwa() {
	return "mlecz";
}

File: Projekt2_java/src/projekt2/rosliny/Mlecz.java
package projekt2.rosliny;

import projekt2.*;

import java.awt.*;

public class Mlecz extends Roslina {

    public Mlecz(Pair pole, Swiat swiat) {
        super(0, 0, pole, swiat);
    }

    @Override
    public Organizm potomek(Pair pole) {
        return new Mlecz(pole, swiat);
    }

    @Override
    public void akcja() {
        rozprzestrzenianie();
        rozprzestrzenianie();
        rozprzestrzenianie();
    }

    @Override
    public Color rysuj() {
        return Color.YELLOW.brighter();
    }

    @Override
    public String getNazwa() {
        return "mlecz";
    }
}
